### DISCUSSION TITLE
Tailwind compatible libraries?

### DISCUSSION THREAD

**Discussion by nikkwong** [expo/router#265] [Q&A]
Since expo-router depends on Metro (which doesn't support CSS), the popular Nativewind library is not compatible with this library. I've tried `twrnc` which works, but does not support web properly (see [this](https://github.com/jaredh159/tailwind-react-native-classnames/discussions/217)). I am wondering if anyone is using any tailwind libraries that are fully compatible with expo-router. Thanks.

**Comment by EvanBacon** [comment]
Which CSS features are required? If simple global CSS support works then I can enable it by merging this https://github.com/expo/router/pull/223

**Reply by nikkwong** [reply]
That would be amazing. Nativewind seems metro to support the bundling of CSS to work properly--therefore it's incompatible with expo-router since expo-router is dependent on Metro. From their [docs](https://next.nativewind.dev/quick-starts/expo-web):

```
NativeWind does not support:

SDK <=45: NativeWind requires React Native Web 0.18 which was added in SDK 46.
Expo Metro: As it [does not support bundling CSS](https://docs.expo.dev/guides/customizing-metro/#expo-webpack-vs-expo-metro)
Expo Router: It requires Expo Metro
```

I'd opt for any library that supports tailwind, however I can't find one that has full support for all basic features of Tailwind other than Nativewind. Thank you!

**Reply by derek-primumco** [reply]
If it helps the discussion at all, the [Solito / Tailwind CSS starter] uses a single [`global.css` file](https://github.com/nandorojo/solito/blob/master/example-monorepos/with-tailwind/apps/next/global.css) in the Next.js app along with this [`tailwind.config.js` file](https://github.com/nandorojo/solito/blob/master/example-monorepos/with-tailwind/apps/next/tailwind.config.js).
